File: pset1/Greedy/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //Uses float as type of change
    float change;
    printf("O hai! ");
    
    /* 
    asks the user for the amount of change that is owned 
    as long as the change is les than 0 
    */
    do
    {
        printf("How much change is owed?\n");
        change = GetFloat();
    }
    while(change < 0);
    
    int cents = round(change * 100);
    int coins = 0;
    /*
    as long as cent is greater than 0 run the code
    to calculate the amount of coin
    */
    while(cents > 0)
    {
        if(cents >= 25)
        {
            cents -= 25;
            coins++;
        }
        else if(cents >= 10)
        {
            cents -= 10;
            coins++;
        }
        else if(cents >= 5)
        {
            cents -= 5;
            coins++;
        }
        else
        {
            cents--;
            coins++;
        }
    }
    printf("%i\n", coins);
}

File: pset6/greedy/greedy.py
import sys
def main():
    #Sets change to float
    change = 0.0;
    print("O hai! ");
    
    while (change < 0):
        print ("How much change is owed?\n");
        #Ask user to input change amount
        change = input("Input change amount: ")
        
    cents = round(change * 100);
    coins = 0;
        
    while(cents > 0):
        if (cents >= 25):
            cents -= 25
            coins = coins + 1
        elif(cents >= 10):
            cents -= 10
            coins = coins + 1
        elif(cents >= 5):
            cents -= 5;
            coins = coins + 1
        else:
            cents = cents - 1
            coins = coins - 1
    
    print(coins)
    print("\n")

    
        